client.go

```diff
@@ -229,34 +230,37 @@ func (c Client) handleResponse(req *internalRequest, response *http.Response, in
 	return nil
 }
 
-func contains(slice []int, val int) bool {
-	for _, item := range slice {
-		if item == val {
-			return true
-		}
-	}
-	return false
+// defaultWaitForPendingUpdate checks each 50ms the status of a WaitForPendingUpdate.
+// This method is used for test purpose
+func (c Client) defaultWaitForPendingUpdate(indexUID string, updateID *AsyncUpdateID) (UpdateStatus, error) {
+	ctx, cancelFunc := context.WithTimeout(context.Background(), time.Second*5)
+	defer cancelFunc()
+
+	return c.WaitForPendingUpdate(ctx, time.Millisecond*50, indexUID, updateID)
 }
 
-// AwaitAsyncUpdateID check each 16ms the status of a AsyncUpdateID.
-// This method should be avoided.
-// TODO: improve this method by returning a channel
-func (c Client) AwaitAsyncUpdateID(indexID string, updateID *AsyncUpdateID) UpdateStatus {
+// WaitForPendingUpdate waits for the end of an update.
+// The function will check by regular interval provided in parameter interval
+// the UpdateStatus. If it is not UpdateStatusEnqueued or the ctx cancelled
+// we return the UpdateStatus.
+func (c Client) WaitForPendingUpdate(
+	ctx context.Context,
+	interval time.Duration,
+	indexID string,
+	updateID *AsyncUpdateID) (UpdateStatus, error) {
+
```

Comment:
I don't find it natural not to be able to use this method only with context. What do you think about just creating the `WaitForPendingUpdate` with the content of the `defaultWaitForPendingUpdate `. We can take a radical choice and force the interval to 1 second and the timeout to 1 minute. In the future, we will probably add a channel to receive updates status. 
What do you think @alexisvisco @curquiza @eskombro?

Reply:
I'm not really comfortable at all with the idea of providing a function with a timeout that the user can not manipulate. Not everyone accepts to wait for 1 minute for example. This kind of function **must** be customizable for me.

But if you want guys, we can pass the function `defaultWaitForPendingUpdate` in public again? In this case, sorry @alexisvisco for the useless change in the previous PR 😅
The user would have 2 choices: the default one with `defaultWaitForPendingUpdate`, and the customizable one with `WaitForPendingUpdate`.

Reply:
(Plus, that's weird that this change is in this PR, because I merged it in the previous one. Can you rebase please? 🙂)

Reply:
I agree, it is important to be able to customize a timeout. Could it be a field in the Client config struct? So we have a default value for a client, no need for params, but can be modified by user?

Reply:
We thought about this default `timeout` value with Charlotte too 👍 
It might come with the next big changes, but not a priority for the moment. We'll apply that in all the SDK in the same time :)

Reply:
(Except if someone has time to do it of course haha)

Reply:
Then it might be an option, as you said, to export both functions (`defaultWaitForPendingUpdate` => `DefaultWaitForPendingUpdate`) while some customization options/methods are actually integrated

Reply:
If we provide a way to define the interval and the timeout in the client this will confuse the user: 

- he may think timeout is for all http requests
- he may think why the fuck there is an interval option in the client while this parameter is used only for one function.

I will say again what I said in the pr: the user can create his own function with default parameters if he want.

If the library provide default values for the `WaitForPendingUpdate` then it will be necessary to experiment to take the best values according some benchmarks. It may be a waste of time to worry about it.

Reply:
Could possibly solve all this questions:

```go
client.WaitForPendingUpdate("indexId", updateID, withWaithForPendingUpdateContext(ctx), withWaitForPendingUpdateInterval(1 * time.seconds))
```

Functional option parameter is widely used in go and is a good practice. Here if the user does not provide withWaithForPendingUpdateContext or withWaitForPendingUpdateInterval: default values will be set. 

Why this painful naming: withWaithForPendingUpdate... ? 
Because if the client take function option parameter in the future I prefer let the name `with` + `the name of the option` to the client.

Reply:
> he may think why the fuck there is an interval option in the client while this parameter is used only for one function.

Don't worry, it's only about the timeout in the client, not the interval 😉
But indeed, the "general" timeout (defined in the client) is not necessary the time to wait for the update but more for the HTTP requests. So it makes no sense to define the timeout for the update in the client, right.

> Functional option parameter is widely used in go and is a good practice. Here if the user does not provide withWaithForPendingUpdateContext or withWaitForPendingUpdateInterval: default values will be set.

I'm glad to see there is a possibility to call a function without some parameters in go! That's what I'm asking since the beginning!! 😂
So, I can use the function with this syntax: `client.WaitForPendingUpdate("indexUID", updateID)` without specify the two last parameters, and you confirm it will work with the default values? @alexisvisco

Reply:
Yes you can: https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
Hi, sorry for the confusion. We've had a long talk with the team and we like the basic idea of having a `WaitForPendingUpdate` which is configurable and a `DefaultWaitForPendingUpdate` with default configuration as you write before. I'm sorry again for all the changes. If you want to talk about it in real-time, I would like to invite you to our [slack community](slack.meilisearch.com) 😃.

Reply:
@qdequele You want the Default method to be public ?

Reply:
Yes, please 😀. And again, thanks a lot for your work!